src/Kentico.Xperience.Lucene/IndexStore.cs

```diff
@@ -36,33 +37,29 @@ public void AddIndex(LuceneIndex index)
             throw new InvalidOperationException($"Attempted to register Lucene index with name '{index.IndexName},' but it is already registered.");
         }
 
-        index.Identifier = registeredIndexes.Count + 1;
         registeredIndexes.Add(index);
     }
 
-
-    /// <summary>
-    /// Adds a crawler to the store.
-    /// </summary>
-    /// <param name="crawlerId">The ID of the crawler to add.</param>
-    /// <exception cref="ArgumentNullException" />
-    /// <exception cref="InvalidOperationException" />
-    public void AddCrawler(string crawlerId)
+    public void AddIndices(IEnumerable<LuceneConfigurationModel> models)
     {
-        if (string.IsNullOrEmpty(crawlerId))
-        {
-            throw new ArgumentNullException(crawlerId);
-        }
-
-        if (registeredCrawlers.Any(id => id.Equals(crawlerId, StringComparison.OrdinalIgnoreCase)))
+        registeredIndexes.Clear();
+        foreach (var index in models)
         {
-            throw new InvalidOperationException($"Attempted to register Lucene crawler with ID '{crawlerId},' but it is already registered.");
+            index.StrategyName ??= "";
+
+            Instance.AddIndex(new LuceneIndex(
+                new StandardAnalyzer(LuceneVersion.LUCENE_48),
+                index.IndexName ?? "",
+                index.ChannelName ?? "",
+                index.LanguageNames?.ToList() ?? new(),
+                index.Id,
+                index.Paths ?? new(),
+                indexPath: null,
+                luceneIndexingStrategy: (ILuceneIndexingStrategy)(Activator.CreateInstance(StrategyStorage.Strategies[index.StrategyName]) ?? new DefaultLuceneIndexingStrategy())
```

Comment:
These indexing strategy classes are instantiated using `Activator.CreateInstance` which means it won't support DI and requires the use of `Service.Resolve<T>()`.

If instead we resolved an instance of this type when we need to process an item event (ex: in `DefaultLuceneTaskLogger.HandleEvent`) then we could use DI.

We'd store the `Type` of the `ILuceneIndexingStrategy` in the `LuceneIndex` instance and use and injected `IServiceProvider` in `DefaultLuceneTaskLogger` to get an instance of the strategy.

Reply:
I updated the implementation to store the `Type` and not an instance of the `ILuceneIndexingStrategy`. The instance is generated dynamically at runtime and now supports DI.